### PR TITLE
Removed `dk` and `lt` prefixes from strings in code

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
- replaced `dk` and `lt` with `dark_` and `light_` within all strings in code;
- removed `dk` and `lt` serach and replace from `color_from_string` and `bgcolor_from_string` functions.

### CODE REVIEW

File: src/color.cpp [hunk 1]
```diff
@@ -530,7 +530,7 @@ nc_color color_from_string(const std::string &color)
         new_color = "c_" + new_color;
     }
 
-    const std::pair<std::string, std::string> pSearch[2] = {{"light_", "lt"}, {"dark_", "dk"}};
+    const std::pair<std::string, std::string> pSearch[2] = {{"light_"}, {"dark_"}};
```
Comment (OzoneH3) [inline src/color.cpp]: This needs to be removed, it converted light_ and dark_ back to lt and dk.

File: src/color.cpp [hunk 2]
```diff
@@ -573,7 +573,7 @@ nc_color bgcolor_from_string(std::string color)
 {
     color = "i_" + color;
 
-    const std::pair<std::string, std::string> pSearch[2] = {{"light_", "lt"}, {"dark_", "dk"}};
+    const std::pair<std::string, std::string> pSearch[2] = {{"light_"}, {"dark_"}};
```
Comment (OzoneH3) [inline src/color.cpp]: Same as above.

File: src/color.cpp (hunk 1 above)
Comment (ZhilkinSerg) [reply src/color.cpp]: Yes, I missed that.

Comment (OzoneH3) [reply src/color.cpp]: Or swap them so it converts the other way round for any stray lt and dk that come in from external mods for compatibility.

File: src/color.cpp [hunk 3]
```diff
@@ -210,157 +210,157 @@ void color_manager::load_default()
     //        Color         Name      Color Pair      Invert
     add_color( def_c_black, "c_black", color_pair( 30 ), def_i_black );
     add_color( def_c_white, "c_white", color_pair( 1 ).bold(), def_i_white );
-    add_color( def_c_ltgray, "c_ltgray", color_pair( 1 ), def_i_ltgray );
-    add_color( def_c_dkgray, "c_dkgray", color_pair( 30 ).bold(), def_i_dkgray );
+    add_color( def_c_ltgray, "c_light_gray", color_pair( 1 ), def_i_ltgray );
```
Comment (OzoneH3) [inline src/color.cpp]: Probably should change the enum def_c_* as well to be consistend.

Comment (ZhilkinSerg) [reply src/color.cpp]: Will address this in subsequent PR #22572.

**Comment by kevingranade** [comment]
This is going to break a lot of mods, the grey/gray thing was just a preview.  Using the lt/dk variants needs to be treated as a warning for at least a month or so to give mod authors time to update without breaking users in the meantime.

We need a mechanism that will break the build if the deprecated versions are used, but will still allow the game to start correctly if a mod uses the old version of the color strings.  If you're not up to that, just revert the change to color_from_string() and bgcolor_from_string() and the rest of this PR (and the next one) is fine to merge.

**Comment by ZhilkinSerg** [comment]
> This is going to break a lot of mods, the grey/gray thing was just a preview. Using the lt/dk variants needs to be treated as a warning for at least a month or so to give mod authors time to update without breaking users in the meantime.

Will following message in the debug.log be enough for that?

    7:43:9.663 WARNING : Deperecated foregorund color suffix was used: dk in color string: dkgray.  
    Please update mod that uses that.

**Comment by AMurkin** [comment]
> Deperecated foregorund

That's serious.

**Comment by AMurkin** [comment]
Oh, and spaces after/before brackets.

**Comment by ZhilkinSerg** [comment]
Updated message format:

    7:43:9.663 WARNING : Deperecated background color suffix was used: (lt) in (ltgrey).  Please update mod that uses that.

**Pull request merged**